docs/smart-contracts/guides/conventions/comprehensive-authentication.mdx

```diff
@@ -0,0 +1,549 @@
+
+#### Authorization Workflow
+
+1. **Invocation Structure**: In Soroban, every transaction can be viewed as a tree of invocations, which are calls to contract functions. Each invocation can generate sub-invocations, creating a layered structure of calls.
+
+2. **Authorization Checks**: Authorization checks are performed at these invocation boundaries. This structure is crucial because each contract may have different authors with varying trust levels. The boundaries where these invocations occur are logical points to perform authorization checks.
+
+3. **Requiring Authorization**: At any point during an invocation, a contract may call the `require_auth(Address)` function provided by the host. This function is designed to ensure that the invoking entity has the necessary authorization to execute the intended action.
+
+4. **Dynamic Authorization**: Soroban allows contracts to not only require but also provide authentication and authorization services dynamically. This flexibility supports complex architectures where contracts can act based on real-time decisions regarding permissions.
+
+#### Overview of Soroban Authentication
+
+Soroban smart contracts allow for complex authorization mechanisms. This guide will help you implement a custom account contract with multi-signature support and demonstrate how to use various authentication functions like `require_auth`, `require_auth_for_args`, and `authorize_as_current_contract`. You can also implement custom authentication logic by implementing `__check_auth()` as a function on your contract.
+
+#### `require_auth`
+
+Certainly! Here's a revised version of the `require_auth` section with a discussion on the implications of omitting this call:
+
+---
+
+#### `require_auth`
+
+The `require_auth` function is used to ensure that a particular address has authorized the current contract invocation. This function is for enforcing that an action within a contract has the explicit permission of the specified address. By calling `require_auth`, you can guarantee that the caller has the necessary authorization to execute the intended operation.
+
+**Example Usage:**
+
+```rust
+impl IncrementContract {
+    pub fn increment(env: Env, user: Address, value: u32) -> u32 {
+        user.require_auth(); // Ensure the user has authorized this call
+        value + 1 // Proceed with increment logic
+    }
+}
+```
+
+**Implications of Omitting `require_auth`:**
+
+If you omit the `require_auth` call in your contract functions, the function can be executed without any authorization from the specified address. This means that anyone could potentially invoke these functions, leading to unauthorized actions. For example, without `require_auth`, an unauthorized user could increment a value or transfer tokens without the owner's consent. Therefore, it is important to include `require_auth` wherever authorization is needed to prevent unauthorized access and maintain the security of your smart contract.
```

Comment:
```suggestion
If you omit the `require_auth` call in your contract functions, the function can be executed without any authorization from the specified address. This means that anyone could potentially invoke these functions, leading to unauthorized actions. For example, without `require_auth`, an unauthorized user could increment a value or transfer tokens without the owner's consent. Therefore, it is important to include `require_auth` wherever authorization is needed to prevent unauthorized access and maintain the security of your smart contract.
```

> For example, without `require_auth`, an unauthorized user could increment a value or transfer tokens without the owner's consent.

This isn't true for the SAC because it does `require_auth` internally, so maybe clarify that?

Reply:
> I second this. It's important to understand the footprint side of auth and sub invocations. I would actually like to see a whole section probably prior to the contract side of this doc on how the footprint plays into this. Both for root invocations as well as sub invocations like in the case of SACs or other cross contract calls which implement auth.

I thought I had covered the footprint in the part called authorization vector but ill revisit it. Thanks.